source/fundamentals/enterprise-authentication.txt

```diff
@@ -2,4 +2,261 @@
+
+- Realm
+
+  The following example shows how to specify the user's realm when it is different
+  from the service's realm:
+  
+  .. tabs::
+  
+     .. tab:: Connection String
+        :tabid: connection-string
+  
+        .. code-block:: csharp
+  
+           var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>/?authMechanism=GSSAPI&authMechanismProperties=REALM:<user's realm>");
+  
+     .. tab:: MongoCredential
+        :tabid: mongo-credential
+  
+        .. code-block:: csharp
+           :emphasize-lines: 1, 2
+  
+           var credential = MongoCredential.CreateGssapiCredential("<username>", "<password>");
+           credential = credential.WithMechanismProperty("REALM", "<user's realm>");
+           
+           var settings = MongoClientSettings.FromConnectionString("<connection string>");
+           settings.Credential = credentials;
+           var mongoClient = new MongoClient(settings);
+
+- Service name
+
+  The following example shows how to specify the service name when it is not the
+  default ``mongodb``:
+  
+  .. tabs::
+  
+     .. tab:: Connection String
+        :tabid: connection-string
+  
+        .. code-block:: csharp
+  
+           var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>/?authMechanism=GSSAPI&authMechanismProperties=SERVICE_NAME:<service name>");
+  
+     .. tab:: MongoCredential
+        :tabid: mongo-credential
+  
+        .. code-block:: csharp
+           :emphasize-lines: 1, 2
+  
+           var credential = MongoCredential.CreateGssapiCredential("<username>", "<password>");
+           credential = credential.WithMechanismProperty("SERVICE_NAME", "<service name>");
+           
+           var settings = MongoClientSettings.FromConnectionString("<connection string>");
+           settings.Credential = credentials;
+           var mongoClient = new MongoClient(settings);
+
+- Multiple properties
+  
+  The following example shows how to specify multiple authentication mechanism
+  properties:
+
+  .. tabs::
+  
+     .. tab:: Connection String
+        :tabid: connection-string
+  
+        .. code-block:: csharp
+  
+           var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>/?authMechanism=GSSAPI&authMechanismProperties=SERVICE_NAME:<service name>,REALM:<user's realm>");
+  
+     .. tab:: MongoCredential
+        :tabid: mongo-credential
+  
+        .. code-block:: csharp
+           :emphasize-lines: 2, 3
+  
+           var credential = MongoCredential.CreateGssapiCredential("<username>", "<password>");
+           credential = credential.WithMechanismProperty("REALM", "<user's realm>")
+               .WithMechanismProperty("SERVICE_NAME", "<service name>");
+           
+           var settings = MongoClientSettings.FromConnectionString("<connection string>");
+           settings.Credential = credentials;
+           var mongoClient = new MongoClient(settings);
+
+.. _csharp-LDAP:
+
+LDAP (PLAIN)
```

Comment:
Q: Is it common to write "PLAIN" in all caps? (Also, is there another kind of LDAP such that we need to disambiguate here?)

Edit: I see in the code that we use "PLAIN" to indicate LDAP. Would it make more sense to make "LDAP" the heading and then explain later, as you did, why we use "PLAIN" to signify that?

Reply:
I think I could go either way on this. In the original docs, as well as in Node and Java it's specified as LDAP (PLAIN), so I lean a little more that direction for consistency's sake, but if you feel strongly that it's better to remove PLAIN from the header I'm okay with it

Reply:
Gotcha. We should definitely follow the other docs.